src/algebra/category/Module/change_of_rings.lean

```diff
@@ -0,0 +1,74 @@
+/-
+Copyright (c) 2022 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+import algebra.category.Module.basic
+
+/-!
+# Change Of Rings
+
+## Main definitions
+
+* `category_theory.Module.restrict_scalars.functor`: given rings `R, S` and a ring homomorphism
+  `R ⟶ S`, then `restrict_scalars.functor : Module S ⥤ Module R` is defined by `M ↦ M` where
+  `M : S-module` is seen as `R-module` by `r • m := f r • m` and `S`-linear map `l : M ⟶ M'` is
+  `R`-linear as well.
+-/
+
+
+namespace category.Module
+
+namespace restrict_scalars
+
+universes u₁ u₂
+
+variables {R : Type u₁} {S : Type u₂} [ring R] [ring S] (f : R →+* S)
+variable (M : Module S)
+
+/-- Any `S`-module M is also an `R`-module via a ring homomorphism `f : R ⟶ S` by defining
+    `r • m := f r • m` (`module.comp_hom`). This is called restriction of scalars. -/
+def obj' : Module R :=
+{ carrier := M,
+  is_add_comm_group := infer_instance,
+  is_module := module.comp_hom M f }
+
+section
+include f
+
```

Comment:
This is no longer needed since you don't use tactic mode for `has_smul` anymore.
```suggestion
```